airflow/cli/cli_parser.py

```diff
@@ -897,6 +897,13 @@ def string_lower_type(val):
     help="The hostname of job(s) that will be checked.",
 )
 
+ARG_JOB_HOSTNAME_CALLABLE_FILTER = Arg(
+    ("--use-hostname-callable",),
```

Comment:
Not sure I love `use-hostname-callable`. Maybe `local-only`? @dstandish @ephraimbuddy

Reply:
It's a nice suggestion @jedcunningham. I don't like so much `use-hostname-callable`, but I am not sure `local-only` would be very _telling_.

Reply:
What about not adding a new arg but having the hostname arg default to `get_hostname`?
i.e
```python
ARG_JOB_HOSTNAME_FILTER = Arg(
    ("--hostname",),
    default=get_hostname(),
    type=str,
    help="The hostname of job(s) that will be checked.",
)
```
The effect of this would be that the CLI check command would always check with hostname instead of none.

As per the name, what do you think about `--use-configured-hostname` or `--use-default-hostname`

Reply:
Sounds very good, thanks for the proposal. Going to implement this